### PR TITLE
Add generator sample

### PR DESCRIPTION
**Pull request by Colstuwjx**
As our documentation was missing the snmp generator usage, I introduced one.
Thx.

### CODE REVIEW

**Comment by brian-brazil** [comment]
That's for the PR, this is more guide material than reference material which belongs in a README. It's not sane to give users a primer on MIBs and ASN.1 in a basic operational/reference doc, I'd suggest you write a blog post instead.

**Comment by Colstuwjx** [comment]
OK, thank you for your reply.

**Comment by SuperQ** [comment]
I think this would be useful as a separate "Tutorial.md" file linked from the README.  This way it's easier to find.

**Comment by RichiH** [comment]
Tossing it into a howto can't hurt.

Richard

Sent by mobile; excuse my brevity and the wall of text Gmail appends by
default.

**Comment by Colstuwjx** [comment]
Yes, I have been struggling to the generator usage for a while, and a simple tutorial could indeed help us.

**Comment by Colstuwjx** [comment]
Okay.. As this PR seems not able to merge into upstream, I'd like to write some blogs instead.
Thanks.

**Comment by SuperQ** [comment]
Sorry, I didn't notice you had updated the PR, I'm happy to review and merge.

File: generator/README.md [hunk 1]
```diff
@@ -22,6 +22,10 @@ The generator reads in from `generator.yml` and writes to `snmp.yml`.
 
 Additional command are available for debugging, use the `help` command to see them.
 
+## HowTo
+
+There is a simple [tutorial](https://github.com/prometheus/snmp_exporter/tree/master/generator/Tutorial.md) for using generator to generate `snmp.yml` from `generator.yml`.
```
Comment (SuperQ) [inline generator/README.md]: This link should be relative, so just `[tutorial](Tutorial.md)`.

File: generator/Tutorial.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
+
+# SNMP Exporter Config Generator Tutorial
+
+This tutorial explains how to config `generator.yml` uses NetSNMP to parse MIBs which we indeed required, and generates `snmp.yml` config for the snmp_exporter using them.
```
Comment (SuperQ) [inline generator/Tutorial.md]: Maybe 
```
... how the config file `generator.yml` ...
```

Comment (SuperQ) [inline generator/Tutorial.md]: Remove `which we indeed required,`

**Comment by brian-brazil** [comment]
This level of documentation doesn't really belong here, a separate blog post would be best.

**Comment by SuperQ** [comment]
I disagree, it would be very good to have complete documentation with the code.

**Comment by brian-brazil** [comment]
A guide for one particular user's setup is not reference documentation.

**Comment by RichiH** [comment]
Anything at all which increases documentation coverage is a net benefit. Yesterday, two different networking people told me of the workaround they built to get data into Prometheus because they simply gave up on snmp_exporter.

**Comment by SuperQ** [comment]
Yup, the snmp_exporter documentation is not good as-is, an example tutorial is necessary, as the usability is very poor.

**Comment by brian-brazil** [comment]
Not all documentation is automatically a benefit. What's really needed is something that takes users through how the exporter works, then SNMP/MIBs and then the generator.

That type of training does not belong in this repo, I want to stick to reference docs and small examples as we have elsewhere.

**Comment by Colstuwjx** [comment]
Em...  I will correct the tutorial grammar mistakes, and `robustperception.io` maybe a better location for that.

**Comment by Colstuwjx** [comment]
I have made this tutorial more clearly. Please feel happy to merge it if we plan to put that on Github, or just close that, and place these on `robustperception.io` site.

I will also write some blogs in my spare time, to explain how to use `snmp exporter`, even the other prometheus components. But I'm not sure the publish time, haha..

**Comment by Colstuwjx** [comment]
I'd like to close this PR as there is already a [blog post](http://devopstarter.info/snmp-exporter-generator-tutorial/) here to explains how to use the snmp-exporter generator!
Thanks.

**Comment by brian-brazil** [comment]
Great, thanks! I've posted it on the subreddit so others will find it.

There's something odd with the colours in the 1st and 2nd `<pre>` blocks.

**Pull request closed without merging**